<launch>
  <arg name="detected_objects_range" default="/detection/lidar_detector/objects"/>
  <arg name="detected_objects_vision" default="/casper_auto/detection/lidar_detector/objects"/>
  <arg name="sync_topics" default="true"/>
  <arg name="distanceThreshold" default="1.3"/>  <!--距离超过1.3米的两个物体就不认为会有是一个物体的可能性-->
  <arg name="overlap_threshold" default="0.5"/>
  <arg name="namespace" default="/detection/fusion_tools"/>
  <arg name="viz" default="false"/>

  <node name="lidar_object_fusion" pkg="lidar_object_fusion" type="lidar_object_fusion" output="screen">
    <param name="detected_objects_range" value="$(arg detected_objects_range)"/>
    <param name="detected_objects_vision" value="$(arg detected_objects_vision)"/>
    <param name="sync_topics" value="$(arg sync_topics)"/>
    <param name="distanceThreshold" value="$(arg distanceThreshold)"/>
    <param name="overlap_threshold" value="$(arg overlap_threshold)"/>
  </node>

  <node if="$(arg viz)" pkg="rviz" name="rviz" type="rviz" args="-d $(find center_point)/rviz/centerpoint.rviz">
  </node>

  <node pkg="detected_objects_visualizer" type="visualize_detected_objects" name="range_fusion_visualization_01"
        output="screen" ns="$(arg namespace)">
    <param name="objects_src_topic" value="/objects"/>
  </node>
</launch>